File: 11-ContainerWithMostWater/11-ContainerWithMostWater.js
// Last updated: 6/26/2026, 8:55:18 AM
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    var output = 0;
    for(let i = 0, j = (height.length - 1) ; i < j ; ) {
        let tempArea = 0;
        if(height[i] >= height[j]) {
            tempArea = height[j] * (j - i);
            if(tempArea > output) output = tempArea;
            j--;
        }
        else {
            tempArea = height[i] * (j - i);
            if(tempArea > output) output = tempArea;
            i++;
        }
    }
    return output;
};

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.java
// Last updated: 6/26/2026, 8:55:24 AM
class Solution {
    public int maxArea(int[] height) {
        int final_area=0;
        int l=0;
        int r=height.length-1;
        while(l<=r){
            final_area = Math.max(final_area,(r-l)*(Math.min(height[l],height[r])));
            if(height[l]<height[r])l++;
            else r--;
        }
        return final_area;
    }
}